### PR TITLE
Add compliant and noncompliant examples of python/pytorch-redundant-softmax@v1.0

### PR DESCRIPTION
**Pull request by swaraj-amazon**
*Description of changes:*
* Rules Added: python/pytorch-redundant-softmax@v1.0
* Added compliant and non compliant examples

### CODE REVIEW

File: src/python/detectors/pytorch_redundant_softmax/pytorch_redundant_softmax.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        train_loop(train_dataloader, model, loss_fn, optimizer)
+        test_loop(test_dataloader, model, loss_fn)
+
+    print("Done!")
+# {/fact}
+
+
+# {fact rule=pytorch-redundant-softmax@v1.0 defects=0}
+def pytorch_redundant_softmax_compliant():
+    import torch
```
Comment (linghuiluo) [inline src/python/detectors/pytorch_redundant_softmax/pytorch_redundant_softmax.py:98]: Use the compliant example below. 

import matplotlib.pyplot as plt
--
  |   | import pandas as pd
  |   | import torch
  |   |  
  |   | import torch.nn as nn
  |   | from transformers import BertModel, BertForSequenceClassification
  |   |  
  |   |  
  |   | import default_constants
  |   |  
  |   |  
  |   | class BERT(nn.Module):
  |   |  
  |   | def __init__(self, tokenizer, bert_variant=default_constants.BERT):
  |   | super(BERT, self).__init__()
  |   |  
  |   | self.num_labels = default_constants.num_labels
  |   | self.hidden_dim = default_constants.hidden_dim
  |   | self.dropout_prob = default_constants.dropout_prob
  |   |  
  |   | self.bert = BertModel.from_pretrained(bert_variant)
  |   | self.bert.resize_token_embeddings(len(tokenizer))
  |   |  
  |   | self.dropout = nn.Dropout(self.dropout_prob).cuda()
  |   | self.classifier = nn.Linear(self.hidden_dim, self.num_labels).cuda()
  |   | torch.nn.init.xavier_uniform(self.classifier.weight)
  |   |  
  |   |  
  |   | def forward(
  |   | self,
  |   | text,
  |   | labels,
  |   | attention_mask=None,
  |   | token_type_ids=None
  |   | ):
  |   |  
  |   |  
  |   | outputs = self.bert(
  |   | text,
  |   | attention_mask=attention_mask,
  |   | token_type_ids=token_type_ids
  |   | )
  |   |  
  |   | pooled_output = outputs[1]
  |   |  
  |   | pooled_output = self.dropout(pooled_output)
  |   | logits = self.classifier(pooled_output)
  |   |  
  |   | loss_fct = nn.CrossEntropyLoss(weight=torch.Tensor(
  |   | default_constants.class_weight)).cuda(default_constants.DEVICE)
  |   | loss = loss_fct(logits.view(-1, self.num_labels), labels.view(-1))
  |   |  
  |   | return (
  |   | loss,
  |   | logits,
  |   | )

**Review by san2488 — APPROVED** [review]
Comments NOT addressed. Please create another PR with changes addressing comments.
